server/service/apple_mdm.go

```diff
@@ -721,6 +747,86 @@ func (svc *Service) DeleteMDMAppleConfigProfile(ctx context.Context, profileUUID
 	return nil
 }
 
+func (svc *Service) DeleteMDMAppleDeclaration(ctx context.Context, declUUID string) error {
+	// first we perform a perform basic authz check
+	if err := svc.authz.Authorize(ctx, &fleet.Team{}, fleet.ActionRead); err != nil {
+		return ctxerr.Wrap(ctx, err)
+	}
+
+	// check that Apple MDM is enabled - the middleware of that endpoint checks
+	// only that any MDM is enabled, maybe it's just Windows
+	if err := svc.VerifyMDMAppleConfigured(ctx); err != nil {
+		err := fleet.NewInvalidArgumentError("profile_uuid", fleet.AppleMDMNotConfiguredMessage).WithStatus(http.StatusBadRequest)
+		return ctxerr.Wrap(ctx, err, "check macOS MDM enabled")
+	}
+
+	decl, err := svc.ds.GetMDMAppleDeclaration(ctx, declUUID)
+	if err != nil {
+		return ctxerr.Wrap(ctx, err)
+	}
+
+	if _, ok := mdm_types.FleetReservedProfileNames()[decl.Name]; ok {
+		return &fleet.BadRequestError{
+			Message:     "profiles managed by Fleet can't be deleted using this endpoint.",
+			InternalErr: fmt.Errorf("deleting profile %s is not allowed because it's managed by Fleet", decl.Name),
+		}
+	}
+
+	// TODO: refine our approach to deleting restricted/forbidden types of declarations
+	var d fleet.MDMAppleRawDeclaration
```

Comment:
is this just in case a forbidden declaration was uploaded? If so I think we should delete this and trust our validation, and if the declaration gets in, then allow users to delete it

Reply:
. I think we do need to validate here. We're calling them "forbidden" in the DDM context but is a bit confusing because [this](https://github.com/fleetdm/fleet/blob/f91e1edc1311b0479214e1054fcda956315222ed/server/fleet/apple_mdm.go#L620-L621) also includes what we call "reserved" in other places (like OS updates)Because the request is referenced by uuid, it is possible that an API user could submit a delete request with the uuid that matches the upcoming DDM OS updates declaration, for example.

Reply:
But that would be the reserved name, not the forbidden one! both are different in a sense. Reserved is for Fleet to use, forbidden is a declaration that shouldn't be ever used.

Forbidden should go away very soon, as soon as we support assets.

Reply:
btw, it's okay to leave this for now! we can revisit